lib/src/group_changes.dart

```diff
@@ -32,12 +32,36 @@ class MinMax {
 /// sets as keys in a later map, so we use the keys, and can recover the sets.
 Map<String, List<String>> configSets;
 
+class Result {
+  Result(Map<String, dynamic> data)
```

Comment:
I suggest splitting this into a Result(this.previous, this.current, this.expected, this.key, this.matches) constructor and a Result.fromJson(Map<String, dynamic> data) constructor implemented using the primary constructor.